### PR TITLE
Useful small fixes to the registration form

### PR DESCRIPTION
**Pull request by mic4ael**

### CODE REVIEW

File: indico/modules/events/registration/controllers/management/fields.py [hunk 1]
```diff
@@ -130,15 +132,11 @@ class RHRegistrationFormAddField(RHManageRegFormSectionBase):
...
-            if len(date_format) == 2:
-                field_data['time_format'] = date_format[1]
-        elif field_data['input_type'] == 'radio':
-            items = field_data['radioitems']
-            for item in items:
-                item['id'] = unicode(uuid4())
+        try:
+            field_impl_type = get_field_types()[field_data['input_type']]
+            field_impl_type.modify_post_data(field_data)
+        except KeyError:
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/fields.py]: too much code in the try block. only `field_impl_type = get_field_types()[field_data['input_type']]` should be in there.

Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/fields.py]: You also need to run the label-specific code before getting here - labels have no field implementation

File: indico/modules/events/registration/fields/base.py [hunk 2]
```diff
@@ -54,6 +54,10 @@ def create_wtf_field(self):
     def save_data(self, registration, value):
         registration.data.append(RegistrationData(field_data_id=self.form_item.current_data_id, data=value))
 
+    @classmethod
+    def modify_post_data(cls, post_data):
+        return
```
Comment (ThiefMaster) [inline indico/modules/events/registration/fields/base.py]: pass

File: indico/modules/events/registration/fields/simple.py [hunk 3]
```diff
@@ -55,12 +56,19 @@ class SelectField(RegistrationFormFieldBase):
...
-        data = self.form_item.current_data.versioned_data
+        data = self.form_item.data
+        versioned_data = self.form_item.current_data.versioned_data
         try:
-            default_item = data['defaultItem']
+            default_item = data['default_item']
         except KeyError:
             return None
-        return next((x['id'] for x in data['radioitems'] if x['caption'] == default_item), None)
+        return next((x['id'] for x in versioned_data['radioitems'] if x['caption'] == default_item), None)
```
Comment (ThiefMaster) [inline indico/modules/events/registration/fields/simple.py]: We should use the ID and not the caption for the default value.

File: indico/modules/events/registration/controllers/management/fields.py (hunk 1 above)
Comment (mic4ael) [inline indico/modules/events/registration/controllers/management/fields.py]: True, I didn't test it for labels.. Shame on me!

File: indico/modules/events/registration/models/form_fields.py [hunk 4]
```diff
@@ -79,10 +78,7 @@ def view_data(self):
         base_dict.update(is_enabled=self.is_enabled, title=self.title, is_required=self.is_required,
                          input_type=self.input_type, html_name=self.html_field_name,
                          **super(RegistrationFormField, self).view_data)
-        if self.input_type == 'country':
-            base_dict['radioitems'] = []
-            for key, val in CountryHolder.getCountries().iteritems():
-                base_dict['radioitems'].append({'caption': val, 'countryKey': key})
+        base_dict.update(**self.field_impl.view_data)
```
Comment (ThiefMaster) [inline indico/modules/events/registration/models/form_fields.py]: no need to use `**` here

Comment (mic4ael) [inline indico/modules/events/registration/models/form_fields.py]: Ahh, yeah, true.

**Pull request merged**